File: packages/flutter_shared/lib/tools/json_tool.dart
import 'dart:convert';

/// JSON 工具 — 格式化 / 压缩 / 校验
class JsonTool {
  JsonTool._();

  /// 格式化 JSON
  static ({String result, String? error}) format(String input, {int indent = 2}) {
    if (input.trim().isEmpty) {
      return (result: '', error: '请输入 JSON');
    }
    try {
      final parsed = jsonDecode(input);
      final encoder = JsonEncoder.withIndent(' ' * indent);
      return (result: encoder.convert(parsed), error: null);
    } on FormatException catch (e) {
      return (result: input, error: 'JSON 格式错误: ${e.message}');
    }
  }

  /// 压缩 JSON
  static ({String result, String? error}) minify(String input) {
    if (input.trim().isEmpty) {
      return (result: '', error: '请输入 JSON');
    }
    try {
      final parsed = jsonDecode(input);
      return (result: jsonEncode(parsed), error: null);
    } on FormatException catch (e) {
      return (result: input, error: 'JSON 格式错误: ${e.message}');
    }
  }

  /// 校验 JSON
  static ({bool valid, String? error, dynamic parsed}) validate(String input) {
    if (input.trim().isEmpty) {
      return (valid: false, error: '请输入 JSON', parsed: null);
    }
    try {
      final parsed = jsonDecode(input);
      return (valid: true, error: null, parsed: parsed);
    } on FormatException catch (e) {
      return (valid: false, error: e.message, parsed: null);
    }
  }

  /// 示例 JSON
  static String get sample => jsonEncode({
    'name': '工具箱',
    'version': '1.0.0',
    'tools': ['JSON', 'Base64', 'Hash'],
    'enabled': true,
    'count': 3,
    'metadata': {
      'author': 'toolbox',
      'created': '2024-01-15',
    },
  });
}

File: miniapp/src/utils/tools/jsonTool.ts
/**
 * JSON 工具 — 格式化 / 压缩 / 校验
 * 移植自 flutter_shared/tools/json_tool.dart
 */

export interface JsonResult {
  result: string;
  error?: string;
}

export interface JsonValidateResult {
  valid: boolean;
  error?: string;
  parsed?: any;
}

export const JsonTool = {
  format(input: string, indent = 2): JsonResult {
    const trimmed = input.trim();
    if (!trimmed) return { result: '', error: '请输入 JSON' };
    try {
      const parsed = JSON.parse(trimmed);
      return { result: JSON.stringify(parsed, null, indent), error: undefined };
    } catch (e: any) {
      return { result: input, error: `JSON 格式错误: ${e.message}` };
    }
  },

  minify(input: string): JsonResult {
    const trimmed = input.trim();
    if (!trimmed) return { result: '', error: '请输入 JSON' };
    try {
      const parsed = JSON.parse(trimmed);
      return { result: JSON.stringify(parsed), error: undefined };
    } catch (e: any) {
      return { result: input, error: `JSON 格式错误: ${e.message}` };
    }
  },

  validate(input: string): JsonValidateResult {
    const trimmed = input.trim();
    if (!trimmed) return { valid: false, error: '请输入 JSON' };
    try {
      const parsed = JSON.parse(trimmed);
      return { valid: true, parsed };
    } catch (e: any) {
      return { valid: false, error: e.message };
    }
  },

  get sample(): string {
    return JSON.stringify({
      name: '工具箱',
      version: '1.0.0',
      tools: ['JSON', 'Base64', 'Hash'],
      enabled: true,
      count: 3,
      metadata: {
        author: 'toolbox',
        created: '2024-01-15',
      },
    }, null, 2);
  },
};
